AnkiDroid/src/test/java/com/ichi2/anki/ModelFieldEditorTest.java

```diff
@@ -0,0 +1,193 @@
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A      *
+ * PARTICULAR PURPOSE. See the GNU General Public License for more details.             *
+ *                                                                                      *
+ * You should have received a copy of the GNU General Public License along with         *
+ * this program.  If not, see <http://www.gnu.org/licenses/>.                           *
+ ****************************************************************************************/
+
+package com.ichi2.anki;
+
+import android.content.Intent;
+import android.widget.EditText;
+
+import com.afollestad.materialdialogs.MaterialDialog;
+import com.ichi2.anki.exception.ConfirmModSchemaException;
+import com.ichi2.libanki.Models;
+import com.ichi2.testutils.AnkiAssert;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.ParameterizedRobolectricTestRunner;
+import org.robolectric.ParameterizedRobolectricTestRunner.Parameters;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static com.afollestad.materialdialogs.DialogAction.POSITIVE;
+import static com.ichi2.anki.FieldOperationType.ADD_FIELD;
+import static com.ichi2.anki.FieldOperationType.RENAME_FIELD;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.not;
+
+@RunWith(ParameterizedRobolectricTestRunner.class)
+public class ModelFieldEditorTest extends RobolectricTest {
+    private static final String[] mForbiddenCharacters = new String[] {"#", "^", "/", " ", "\t"};
+    private String mForbiddenCharacter;
+
+
+    public ModelFieldEditorTest(String forbiddenCharacter) {
+        this.mForbiddenCharacter = forbiddenCharacter;
+    }
+
+
+    @Parameters
+    public static Iterable<? extends Object> forbiddenCharacters() {
+        return Arrays.asList(mForbiddenCharacters);
+    }
+
+
+    /**
+     * Tests if field names with illegal characters get removed from beginning of field names when adding field
+     */
+    @Test
+    public void testIllegalCharactersInFieldName_addField() {
+        // Assertion fails in case of a ConfirmModSchemaException being thrown
+        AnkiAssert.assertDoesNotThrow(() -> {
+            String fieldName = setupInvalidFieldName(mForbiddenCharacter, ADD_FIELD);
+
+            testForIllegalCharacters(fieldName);
+        });
+    }
+
+
+    /**
+     * Tests if field names with illegal characters get removed from beginning of field names when renaming field
+     */
+    @Test
+    public void testIllegalCharactersInFieldName_renameField() {
+        // Assertion fails in case of a ConfirmModSchemaException being thrown
+        AnkiAssert.assertDoesNotThrow(() -> {
+            String fieldName = setupInvalidFieldName(mForbiddenCharacter, RENAME_FIELD);
+
+            testForIllegalCharacters(fieldName);
+        });
```

Comment:
That code was designed for an overzealous code analyzer that we no longer user

```suggestion
        String fieldName = setupInvalidFieldName(mForbiddenCharacter, RENAME_FIELD);

        testForIllegalCharacters(fieldName);
```

Reply:
I still use codacy ;-), that said, it could always be tweaked - we define(d) it's rule set